app/generic/definitions.py

```diff
@@ -0,0 +1,390 @@
+            sig_type = signature["operation"]["signature_type"]
+            schema_ref = signature["operation"]["ref"]
+            tag = signature["operation"]["tag"]
+            credential_definition_id = f"{public_did}:3:{sig_type}:{schema_ref}:{tag}"
+        except Exception as e:
+            raise CloudApiException(
+                "Unable to construct credential definition id from signature response"
+            ) from e
+
+        # FIXME: ACA-Py 0.7.3 has no way to associate the credential definition id with a transaction record
+        # This methods find the created credential definition for a schema and tag (which is always unique)
+        # credential_definition_id = (
+        #     await get_credential_definition_id_for_schema_and_tag(
+        #         aries_controller,
+        #         credential_definition.schema_id,
+        #         credential_definition.tag,
+        #     )
+        # )
+        # if not credential_definition_id:
+        #     raise CloudApiException(
+        #         f"Could not find any created credential definitions for schema_id {credential_definition.schema_id} and tag {credential_definition.tag}",
+        #         500,
+        #     )
+
+    else:
+        await stop_listener()
+        credential_definition_id = result.credential_definition_id
+
+    # ACA-Py only returns the id after creating a credential definition
+    # We want consistent return types across all endpoints, so retrieving the credential
+    # definition here.
+    return await get_credential_definition_by_id(
+        credential_definition_id, aries_controller
+    )
+
+
+@router.get("/schemas", response_model=List[CredentialSchema])
+async def get_schemas(
+    schema_id: Optional[str] = None,
+    schema_issuer_did: Optional[str] = None,
+    schema_name: Optional[str] = None,
+    schema_version: Optional[str] = None,
+    aries_controller: AcaPyClient = Depends(agent_selector),
+):
+    """
+    Retrieve schemas that the current agent created.
+
+    Parameters:
+    -----------
+    schema_id: str (Optional)
+    schema_issuer_did: str (Optional)
+    schema_name: str (Optional)
+    schema_version: str (Optional)
+
+    Returns:
+    --------
+    Json response with created schemas from ledger.
+    """
+    # Get all created schema ids that match the filter
+    response = await aries_controller.schema.get_created_schemas(
+        schema_id=schema_id,
+        schema_issuer_did=schema_issuer_did,
+        schema_name=schema_name,
+        schema_version=schema_version,
+    )
+
+    # Initiate retrieving all schemas
+    schema_ids = response.schema_ids or []
+    get_schema_futures = [
+        aries_controller.schema.get_schema(schema_id=schema_id)
+        for schema_id in schema_ids
+    ]
+
+    # Wait for completion of retrieval and transform all schemas into response model (if a schema was returned)
+    schema_results = await asyncio.gather(*get_schema_futures)
```

Comment:
Same as above - I think you can just do await in the list comprehension?

Reply:
Same, want to run in parallel